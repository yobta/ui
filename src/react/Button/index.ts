export * from './Button'
export * from './createButton'
