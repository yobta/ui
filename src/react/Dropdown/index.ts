export * from './createDropdown.js'
export * from './Dropdown.js'
