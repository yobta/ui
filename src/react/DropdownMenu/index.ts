export * from './createDropdownMenu.js'
export * from './DropdownMenu.js'
