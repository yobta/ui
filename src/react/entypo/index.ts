export { FiveHundredPx } from './icons/500px.js'
export { FiveHundredWithCircle } from './icons/500pxWithCircle.js'
export { Address } from './icons/Address.js'
export { AddToList } from './icons/AddToList.js'
export { AddUser } from './icons/AddUser.js'
export { Adjust } from './icons/Adjust.js'
export { AircraftLanding } from './icons/AircraftLanding.js'
export { AircraftTakeOff } from './icons/AircraftTakeOff.js'
export { Aircraft } from './icons/Aircraft.js'
export { Air } from './icons/Air.js'
export { AlignBottom } from './icons/AlignBottom.js'
export { AlignHorizontalMiddle } from './icons/AlignHorizontalMiddle.js'
export { AlignLeft } from './icons/AlignLeft.js'
export { AlignRight } from './icons/AlignRight.js'
export { AlignTop } from './icons/AlignTop.js'
export { AlignVerticalMiddle } from './icons/AlignVerticalMiddle.js'
export { AppStore } from './icons/AppStore.js'
export { Archive } from './icons/Archive.js'
export { AreaGraph } from './icons/AreaGraph.js'
export { ArrowBoldDown } from './icons/ArrowBoldDown.js'
export { ArrowBoldLeft } from './icons/ArrowBoldLeft.js'
export { ArrowBoldRight } from './icons/ArrowBoldRight.js'
export { ArrowBoldUp } from './icons/ArrowBoldUp.js'
export { ArrowDown } from './icons/ArrowDown.js'
export { ArrowLeft } from './icons/ArrowLeft.js'
export { ArrowLongDown } from './icons/ArrowLongDown.js'
export { ArrowLongLeft } from './icons/ArrowLongLeft.js'
export { ArrowLongRight } from './icons/ArrowLongRight.js'
export { ArrowLongUp } from './icons/ArrowLongUp.js'
export { ArrowRight } from './icons/ArrowRight.js'
export { ArrowUp } from './icons/ArrowUp.js'
export { ArrowWithCircleDown } from './icons/ArrowWithCircleDown.js'
export { ArrowWithCircleLeft } from './icons/ArrowWithCircleLeft.js'
export { ArrowWithCircleRight } from './icons/ArrowWithCircleRight.js'
export { ArrowWithCircleUp } from './icons/ArrowWithCircleUp.js'
export { Attachment } from './icons/Attachment.js'
export { AwarenessRibbon } from './icons/AwarenessRibbon.js'
export { BackInTime } from './icons/BackInTime.js'
export { Back } from './icons/Back.js'
export { Baidu } from './icons/Baidu.js'
export { BarGraph } from './icons/BarGraph.js'
export { Basecamp } from './icons/Basecamp.js'
export { Battery } from './icons/Battery.js'
export { BeamedNote } from './icons/BeamedNote.js'
export { Behance } from './icons/Behance.js'
export { Bell } from './icons/Bell.js'
export { Blackboard } from './icons/Blackboard.js'
export { Block } from './icons/Block.js'
export { Bookmarks } from './icons/Bookmarks.js'
export { Bookmark } from './icons/Bookmark.js'
export { Book } from './icons/Book.js'
export { Bowl } from './icons/Bowl.js'
export { Box } from './icons/Box.js'
export { Briefcase } from './icons/Briefcase.js'
export { Browser } from './icons/Browser.js'
export { Brush } from './icons/Brush.js'
export { Bucket } from './icons/Bucket.js'
export { Bug } from './icons/Bug.js'
export { Cake } from './icons/Cake.js'
export { Calculator } from './icons/Calculator.js'
export { Calendar } from './icons/Calendar.js'
export { Camera } from './icons/Camera.js'
export { Ccw } from './icons/Ccw.js'
export { Chat } from './icons/Chat.js'
export { Check } from './icons/Check.js'
export { ChevronDown } from './icons/ChevronDown.js'
export { ChevronLeft } from './icons/ChevronLeft.js'
export { ChevronRight } from './icons/ChevronRight.js'
export { ChevronSmallDown } from './icons/ChevronSmallDown.js'
export { ChevronSmallLeft } from './icons/ChevronSmallLeft.js'
export { ChevronSmallRight } from './icons/ChevronSmallRight.js'
export { ChevronSmallUp } from './icons/ChevronSmallUp.js'
export { ChevronThinDown } from './icons/ChevronThinDown.js'
export { ChevronThinLeft } from './icons/ChevronThinLeft.js'
export { ChevronThinRight } from './icons/ChevronThinRight.js'
export { ChevronThinUp } from './icons/ChevronThinUp.js'
export { ChevronUp } from './icons/ChevronUp.js'
export { ChevronWithCircleDown } from './icons/ChevronWithCircleDown.js'
export { ChevronWithCircleLeft } from './icons/ChevronWithCircleLeft.js'
export { ChevronWithCircleRight } from './icons/ChevronWithCircleRight.js'
export { ChevronWithCircleUp } from './icons/ChevronWithCircleUp.js'
export { Circle } from './icons/Circle.js'
export { CircleWithCross } from './icons/CircleWithCross.js'
export { CircleWithMinus } from './icons/CircleWithMinus.js'
export { CircleWithPlus } from './icons/CircleWithPlus.js'
export { CircularGraph } from './icons/CircularGraph.js'
export { Clapperboard } from './icons/Clapperboard.js'
export { ClassicComputer } from './icons/ClassicComputer.js'
export { Clipboard } from './icons/Clipboard.js'
export { Clock } from './icons/Clock.js'
export { Cloud } from './icons/Cloud.js'
export { Code } from './icons/Code.js'
export { Cog } from './icons/Cog.js'
export { Colours } from './icons/Colours.js'
export { Compass } from './icons/Compass.js'
export { ControllerFastBackward } from './icons/ControllerFastBackward.js'
export { ControllerFastForward } from './icons/ControllerFastForward.js'
export { ControllerJumpToStart } from './icons/ControllerJumpToStart.js'
export { ControllerNext } from './icons/ControllerNext.js'
export { ControllerPaus } from './icons/ControllerPaus.js'
export { ControllerPlay } from './icons/ControllerPlay.js'
export { ControllerRecord } from './icons/ControllerRecord.js'
export { ControllerStop } from './icons/ControllerStop.js'
export { ControllerVolume } from './icons/ControllerVolume.js'
export { Copy } from './icons/Copy.js'
export { CreativeCloud } from './icons/CreativeCloud.js'
export { CreativeCommonsAttribution } from './icons/CreativeCommonsAttribution.js'
export { CreativeCommonsNoderivs } from './icons/CreativeCommonsNoderivs.js'
export { CreativeCommonsNoncommercialEu } from './icons/CreativeCommonsNoncommercialEu.js'
export { CreativeCommonsNoncommercialUs } from './icons/CreativeCommonsNoncommercialUs.js'
export { CreativeCommonsPublicDomain } from './icons/CreativeCommonsPublicDomain.js'
export { CreativeCommonsRemix } from './icons/CreativeCommonsRemix.js'
export { CreativeCommonsSharealike } from './icons/CreativeCommonsSharealike.js'
export { CreativeCommonsShare } from './icons/CreativeCommonsShare.js'
export { CreativeCommons } from './icons/CreativeCommons.js'
export { CreditCard } from './icons/CreditCard.js'
export { Credit } from './icons/Credit.js'
export { Crop } from './icons/Crop.js'
export { Cross } from './icons/Cross.js'
export { Cup } from './icons/Cup.js'
export { Cw } from './icons/Cw.js'
export { Cycle } from './icons/Cycle.js'
export { Database } from './icons/Database.js'
export { DialPad } from './icons/DialPad.js'
export { Direction } from './icons/Direction.js'
export { DocumentLandscape } from './icons/DocumentLandscape.js'
export { Documents } from './icons/Documents.js'
export { Document } from './icons/Document.js'
export { DotSingle } from './icons/DotSingle.js'
export { DotsThreeHorizontal } from './icons/DotsThreeHorizontal.js'
export { DotsThreeVertical } from './icons/DotsThreeVertical.js'
export { DotsTwoHorizontal } from './icons/DotsTwoHorizontal.js'
export { DotsTwoVertical } from './icons/DotsTwoVertical.js'
export { Download } from './icons/Download.js'
export { Dribbble } from './icons/Dribbble.js'
export { DribbbleWithCircle } from './icons/DribbbleWithCircle.js'
export { Drink } from './icons/Drink.js'
export { Drive } from './icons/Drive.js'
export { Dropbox } from './icons/Dropbox.js'
export { Drop } from './icons/Drop.js'
export { Edit } from './icons/Edit.js'
export { Email } from './icons/Email.js'
export { EmojiFlirt } from './icons/EmojiFlirt.js'
export { EmojiHappy } from './icons/EmojiHappy.js'
export { EmojiNeutral } from './icons/EmojiNeutral.js'
export { EmojiSad } from './icons/EmojiSad.js'
export { Eraser } from './icons/Eraser.js'
export { Erase } from './icons/Erase.js'
export { Evernote } from './icons/Evernote.js'
export { Export } from './icons/Export.js'
export { Eye } from './icons/Eye.js'
export { EyeWithLine } from './icons/EyeWithLine.js'
export { Facebook } from './icons/Facebook.js'
export { FacebookWithCircle } from './icons/FacebookWithCircle.js'
export { Feather } from './icons/Feather.js'
export { Fingerprint } from './icons/Fingerprint.js'
export { Flag } from './icons/Flag.js'
export { Flashlight } from './icons/Flashlight.js'
export { Flash } from './icons/Flash.js'
export { FlatBrush } from './icons/FlatBrush.js'
export { Flattr } from './icons/Flattr.js'
export { Flickr } from './icons/Flickr.js'
export { FlickrWithCircle } from './icons/FlickrWithCircle.js'
export { FlowBranch } from './icons/FlowBranch.js'
export { FlowCascade } from './icons/FlowCascade.js'
export { Flower } from './icons/Flower.js'
export { FlowLine } from './icons/FlowLine.js'
export { FlowParallel } from './icons/FlowParallel.js'
export { FlowTree } from './icons/FlowTree.js'
export { FolderImages } from './icons/FolderImages.js'
export { FolderMusic } from './icons/FolderMusic.js'
export { Folder } from './icons/Folder.js'
export { FolderVideo } from './icons/FolderVideo.js'
export { Forward } from './icons/Forward.js'
export { Foursquare } from './icons/Foursquare.js'
export { Funnel } from './icons/Funnel.js'
export { GameController } from './icons/GameController.js'
export { Gauge } from './icons/Gauge.js'
export { Github } from './icons/Github.js'
export { GithubWithCircle } from './icons/GithubWithCircle.js'
export { Globe } from './icons/Globe.js'
export { GoogleDrive } from './icons/GoogleDrive.js'
export { GoogleHangouts } from './icons/GoogleHangouts.js'
export { GooglePlay } from './icons/GooglePlay.js'
export { Googleplus } from './icons/Googleplus.js'
export { GoogleplusWithCircle } from './icons/GoogleplusWithCircle.js'
export { GraduationCap } from './icons/GraduationCap.js'
export { Grid } from './icons/Grid.js'
export { Grooveshark } from './icons/Grooveshark.js'
export { HairCross } from './icons/HairCross.js'
export { Hand } from './icons/Hand.js'
export { HeartOutlined } from './icons/HeartOutlined.js'
export { Heart } from './icons/Heart.js'
export { Help } from './icons/Help.js'
export { HelpWithCircle } from './icons/HelpWithCircle.js'
export { Home } from './icons/Home.js'
export { HourGlass } from './icons/HourGlass.js'
export { Houzz } from './icons/Houzz.js'
export { Icloud } from './icons/Icloud.js'
export { ImageInverted } from './icons/ImageInverted.js'
export { Images } from './icons/Images.js'
export { Image } from './icons/Image.js'
export { Inbox } from './icons/Inbox.js'
export { InfinityIcon as Infinity } from './icons/Infinity.js'
export { Info } from './icons/Info.js'
export { InfoWithCircle } from './icons/InfoWithCircle.js'
export { Instagram } from './icons/Instagram.js'
export { InstagramWithCircle } from './icons/InstagramWithCircle.js'
export { Install } from './icons/Install.js'
export { Keyboard } from './icons/Keyboard.js'
export { Key } from './icons/Key.js'
export { LabFlask } from './icons/LabFlask.js'
export { Landline } from './icons/Landline.js'
export { Language } from './icons/Language.js'
export { Laptop } from './icons/Laptop.js'
export { Lastfm } from './icons/Lastfm.js'
export { LastfmWithCircle } from './icons/LastfmWithCircle.js'
export { Layers } from './icons/Layers.js'
export { Leaf } from './icons/Leaf.js'
export { LevelDown } from './icons/LevelDown.js'
export { LevelUp } from './icons/LevelUp.js'
export { Lifebuoy } from './icons/Lifebuoy.js'
export { LightBulb } from './icons/LightBulb.js'
export { LightDown } from './icons/LightDown.js'
export { LightUp } from './icons/LightUp.js'
export { LineGraph } from './icons/LineGraph.js'
export { Linkedin } from './icons/Linkedin.js'
export { LinkedinWithCircle } from './icons/LinkedinWithCircle.js'
export { Link } from './icons/Link.js'
export { List } from './icons/List.js'
export { LocationPin } from './icons/LocationPin.js'
export { Location } from './icons/Location.js'
export { LockOpen } from './icons/LockOpen.js'
export { Lock } from './icons/Lock.js'
export { Login } from './icons/Login.js'
export { LogOut } from './icons/LogOut.js'
export { Loop } from './icons/Loop.js'
export { Magnet } from './icons/Magnet.js'
export { MagnifyingGlass } from './icons/MagnifyingGlass.js'
export { Mail } from './icons/Mail.js'
export { MailWithCircle } from './icons/MailWithCircle.js'
export { Man } from './icons/Man.js'
export { Map } from './icons/Map.js'
export { Mask } from './icons/Mask.js'
export { Medal } from './icons/Medal.js'
export { Medium } from './icons/Medium.js'
export { MediumWithCircle } from './icons/MediumWithCircle.js'
export { Megaphone } from './icons/Megaphone.js'
export { Menu } from './icons/Menu.js'
export { Merge } from './icons/Merge.js'
export { Message } from './icons/Message.js'
export { Mic } from './icons/Mic.js'
export { Minus } from './icons/Minus.js'
export { Mixi } from './icons/Mixi.js'
export { Mobile } from './icons/Mobile.js'
export { ModernMic } from './icons/ModernMic.js'
export { Moon } from './icons/Moon.js'
export { MousePointer } from './icons/MousePointer.js'
export { Mouse } from './icons/Mouse.js'
export { Music } from './icons/Music.js'
export { Network } from './icons/Network.js'
export { NewMessage } from './icons/NewMessage.js'
export { Newsletter } from './icons/Newsletter.js'
export { News } from './icons/News.js'
export { New } from './icons/New.js'
export { Note } from './icons/Note.js'
export { NotificationsOff } from './icons/NotificationsOff.js'
export { Notification } from './icons/Notification.js'
export { OldMobile } from './icons/OldMobile.js'
export { OldPhone } from './icons/OldPhone.js'
export { Onedrive } from './icons/Onedrive.js'
export { OpenBook } from './icons/OpenBook.js'
export { Palette } from './icons/Palette.js'
export { PaperPlane } from './icons/PaperPlane.js'
export { Paypal } from './icons/Paypal.js'
export { Pencil } from './icons/Pencil.js'
export { Phone } from './icons/Phone.js'
export { Picasa } from './icons/Picasa.js'
export { PieChart } from './icons/PieChart.js'
export { Pinterest } from './icons/Pinterest.js'
export { PinterestWithCircle } from './icons/PinterestWithCircle.js'
export { Pin } from './icons/Pin.js'
export { Plus } from './icons/Plus.js'
export { Popup } from './icons/Popup.js'
export { PowerPlug } from './icons/PowerPlug.js'
export { PriceRibbon } from './icons/PriceRibbon.js'
export { PriceTag } from './icons/PriceTag.js'
export { Print } from './icons/Print.js'
export { ProgressEmpty } from './icons/ProgressEmpty.js'
export { ProgressFull } from './icons/ProgressFull.js'
export { ProgressOne } from './icons/ProgressOne.js'
export { ProgressTwo } from './icons/ProgressTwo.js'
export { Publish } from './icons/Publish.js'
export { Qq } from './icons/Qq.js'
export { QqWithCircle } from './icons/QqWithCircle.js'
export { Quote } from './icons/Quote.js'
export { Radio } from './icons/Radio.js'
export { Raft } from './icons/Raft.js'
export { RaftWithCircle } from './icons/RaftWithCircle.js'
export { Rainbow } from './icons/Rainbow.js'
export { Rdio } from './icons/Rdio.js'
export { RdioWithCircle } from './icons/RdioWithCircle.js'
export { RemoveUser } from './icons/RemoveUser.js'
export { Renren } from './icons/Renren.js'
export { ReplyAll } from './icons/ReplyAll.js'
export { Reply } from './icons/Reply.js'
export { Resize100 } from './icons/Resize100.js'
export { ResizeFullScreen } from './icons/ResizeFullScreen.js'
export { Retweet } from './icons/Retweet.js'
export { Rocket } from './icons/Rocket.js'
export { RoundBrush } from './icons/RoundBrush.js'
export { Rss } from './icons/Rss.js'
export { Ruler } from './icons/Ruler.js'
export { Save } from './icons/Save.js'
export { Scissors } from './icons/Scissors.js'
export { Scribd } from './icons/Scribd.js'
export { SelectArrows } from './icons/SelectArrows.js'
export { Shareable } from './icons/Shareable.js'
export { ShareAlternative } from './icons/ShareAlternative.js'
export { Share } from './icons/Share.js'
export { Shield } from './icons/Shield.js'
export { ShoppingBag } from './icons/ShoppingBag.js'
export { ShoppingBasket } from './icons/ShoppingBasket.js'
export { ShoppingCart } from './icons/ShoppingCart.js'
export { Shop } from './icons/Shop.js'
export { Shuffle } from './icons/Shuffle.js'
export { Signal } from './icons/Signal.js'
export { SinaWeibo } from './icons/SinaWeibo.js'
export { Skype } from './icons/Skype.js'
export { SkypeWithCircle } from './icons/SkypeWithCircle.js'
export { Slideshare } from './icons/Slideshare.js'
export { Smashing } from './icons/Smashing.js'
export { Soundcloud } from './icons/Soundcloud.js'
export { SoundMix } from './icons/SoundMix.js'
export { SoundMute } from './icons/SoundMute.js'
export { Sound } from './icons/Sound.js'
export { SportsClub } from './icons/SportsClub.js'
export { Spotify } from './icons/Spotify.js'
export { SpotifyWithCircle } from './icons/SpotifyWithCircle.js'
export { Spreadsheet } from './icons/Spreadsheet.js'
export { SquaredCross } from './icons/SquaredCross.js'
export { SquaredMinus } from './icons/SquaredMinus.js'
export { SquaredPlus } from './icons/SquaredPlus.js'
export { StarOutlined } from './icons/StarOutlined.js'
export { Star } from './icons/Star.js'
export { Stopwatch } from './icons/Stopwatch.js'
export { Stumbleupon } from './icons/Stumbleupon.js'
export { StumbleuponWithCircle } from './icons/StumbleuponWithCircle.js'
export { Suitcase } from './icons/Suitcase.js'
export { Swap } from './icons/Swap.js'
export { Swarm } from './icons/Swarm.js'
export { Sweden } from './icons/Sweden.js'
export { Switch } from './icons/Switch.js'
export { TabletMobileCombo } from './icons/TabletMobileCombo.js'
export { Tablet } from './icons/Tablet.js'
export { Tag } from './icons/Tag.js'
export { TextDocumentInverted } from './icons/TextDocumentInverted.js'
export { TextDocument } from './icons/TextDocument.js'
export { Text } from './icons/Text.js'
export { Thermometer } from './icons/Thermometer.js'
export { ThumbsDown } from './icons/ThumbsDown.js'
export { ThumbsUp } from './icons/ThumbsUp.js'
export { ThunderCloud } from './icons/ThunderCloud.js'
export { Ticket } from './icons/Ticket.js'
export { TimeSlot } from './icons/TimeSlot.js'
export { Tools } from './icons/Tools.js'
export { TrafficCone } from './icons/TrafficCone.js'
export { Trash } from './icons/Trash.js'
export { Tree } from './icons/Tree.js'
export { TriangleDown } from './icons/TriangleDown.js'
export { TriangleLeft } from './icons/TriangleLeft.js'
export { TriangleRight } from './icons/TriangleRight.js'
export { TriangleUp } from './icons/TriangleUp.js'
export { Tripadvisor } from './icons/Tripadvisor.js'
export { Trophy } from './icons/Trophy.js'
export { Tumblr } from './icons/Tumblr.js'
export { TumblrWithCircle } from './icons/TumblrWithCircle.js'
export { Tv } from './icons/Tv.js'
export { Twitter } from './icons/Twitter.js'
export { TwitterWithCircle } from './icons/TwitterWithCircle.js'
export { Typing } from './icons/Typing.js'
export { Uninstall } from './icons/Uninstall.js'
export { Unread } from './icons/Unread.js'
export { Untag } from './icons/Untag.js'
export { UploadToCloud } from './icons/UploadToCloud.js'
export { Upload } from './icons/Upload.js'
export { Users } from './icons/Users.js'
export { User } from './icons/User.js'
export { VCard } from './icons/VCard.js'
export { VideoCamera } from './icons/VideoCamera.js'
export { Video } from './icons/Video.js'
export { Vimeo } from './icons/Vimeo.js'
export { VimeoWithCircle } from './icons/VimeoWithCircle.js'
export { Vine } from './icons/Vine.js'
export { VineWithCircle } from './icons/VineWithCircle.js'
export { Vinyl } from './icons/Vinyl.js'
export { VkAlternitive } from './icons/VkAlternitive.js'
export { Vk } from './icons/Vk.js'
export { VkWithCircle } from './icons/VkWithCircle.js'
export { Voicemail } from './icons/Voicemail.js'
export { Wallet } from './icons/Wallet.js'
export { Warning } from './icons/Warning.js'
export { Water } from './icons/Water.js'
export { WindowsStore } from './icons/WindowsStore.js'
export { Xing } from './icons/Xing.js'
export { XingWithCircle } from './icons/XingWithCircle.js'
export { Yelp } from './icons/Yelp.js'
export { Youko } from './icons/Youko.js'
export { YoukoWithCircle } from './icons/YoukoWithCircle.js'
export { Youtube } from './icons/Youtube.js'
export { YoutubeWithCircle } from './icons/YoutubeWithCircle.js'
