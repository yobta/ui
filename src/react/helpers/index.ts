export * from './bulk/index.js'
export * from './subscribe/index.js'
export * from './whenOnline/index.js'
export * from './whenTimeout/index.js'
export * from './createUiPortal.js'
