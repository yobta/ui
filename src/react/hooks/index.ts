export * from './useClickAway/index.js'
export * from './useCombineRefs/index.js'
export * from './useConnectionStatus/index.js'
export * from './useDependencyChangeCount/index.js'
export * from './useEscapeKey/index.js'
export * from './useIdRef/index.js'
export * from './useObserveChanges/index.js'
export * from './usePopoverCoordinates/index.js'
export * from './usePatchState.js'
export * from './usePortalNode.js'
