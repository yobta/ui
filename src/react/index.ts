export * from './Button'
export * from './Cache'
export * from './Input'
export * from './LinearProgress'
export * from './PolymorphicComponent'
export * from './Spinner'
export * from './TextArea'

export * from './helpers'
export * from './hooks'
