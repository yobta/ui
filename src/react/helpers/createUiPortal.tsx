import { createContext, ReactNode, ReactPortal } from 'react'
import { createPortal } from 'react-dom'

import { usePortalNode } from '../hooks/usePortalNode/index.js'

type Context = { addToPortal(children: ReactNode): void }

export const context = createContext<Context>({} as Context)

export const createUiPortal = (
  children: ReactNode,
  nodeId?: string
): ReactPortal | ReactNode => {
  let portalNode = usePortalNode(nodeId)

  if (portalNode) {
    return createPortal(children, portalNode)
  }

  return <div className="hidden">{children}</div>
}
