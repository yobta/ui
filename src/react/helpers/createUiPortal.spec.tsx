import { it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import React from 'react'

import { createUiPortal } from './createUiPortal.js'

it('createUiPortal', () => {
  let App = (): JSX.Element => {
    return (
      <>
        {createUiPortal(<span>NewElement</span>, 'nodeid')}
        <div id="nodeid">Element</div>
      </>
    )
  }

  render(<App />)
  let element = screen.getByText<HTMLElement>('Element')

  expect(element).toMatchSnapshot()
})
