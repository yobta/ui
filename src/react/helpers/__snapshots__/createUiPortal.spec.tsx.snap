// Vitest Snapshot v1

exports[`creates a component and passes the node from the portal to it 1`] = `
<div
  id="nodeid"
>
  Element
  <span>
    NewElement
  </span>
</div>
`;
